From c99684477ec66f290ec74efe732acd3c24db7dcc Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 29 Mar 2013 09:16:28 +0400
Subject: [PATCH 06/39] COLLECT_GCC_OPTIONS

This patch adds --sysroot into COLLECT_GCC_OPTIONS which is used to
invoke collect2.

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending
---
 gcc/gcc.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gcc/gcc.c b/gcc/gcc.c
index cd6c6fc95db..7da9c5d457b 100644
--- a/gcc/gcc.c
+++ b/gcc/gcc.c
@@ -4804,6 +4804,15 @@ set_collect_gcc_options (void)
 		sizeof ("COLLECT_GCC_OPTIONS=") - 1);
 
   first_time = TRUE;
+#ifdef HAVE_LD_SYSROOT
+  if (target_system_root_changed && target_system_root)
+    {
+      obstack_grow (&collect_obstack, "'--sysroot=", sizeof("'--sysroot=")-1);
+      obstack_grow (&collect_obstack, target_system_root,strlen(target_system_root));
+      obstack_grow (&collect_obstack, "'", 1);
+      first_time = FALSE;
+    }
+#endif
   for (i = 0; (int) i < n_switches; i++)
     {
       const char *const *args;
-- 
2.25.1

